ibc_public/utils_relaxo.py

```diff
@@ -1,50 +1,81 @@
-                        compute_epi_mask,intersect_masks)
-from nilearn.image import (math_img, resample_to_img, mean_img)
-from nilearn import plotting as nilplot
-from matplotlib.pyplot import (hist, savefig, title, close)
+import time
+from os import listdir, makedirs, sep, system
+from os.path import exists, join
+import numpy as np
 from matplotlib import cm
-from nibabel import load
-from numpy import (nanpercentile, nanmin)
-from scipy import ndimage
+from matplotlib.pyplot import close, hist, savefig, title
+from nibabel import Nifti1Image, load, save
+from nilearn import plotting as nilplot
+from nilearn.image import math_img, mean_img, resample_to_img
 from nilearn.image.image import new_img_like
-
+from nilearn.masking import (compute_background_mask, compute_brain_mask,
+                             compute_epi_mask, intersect_masks)
+from numpy import nanmin, nanpercentile
+from pypreprocess.nipype_preproc_spm_utils import (SubjectData,
+                                                   _do_subject_coregister,
+                                                   _do_subject_normalize,
+                                                   _do_subject_segment)
+from qmri.t2star.trunk.monoexp import monoexponential
+from scipy import ndimage
 
 def closing(image, iterations):
+    """ Wrapper for scipy's binary_closing function to close a NIFTI image with
+    n iterations
+    
+    Parameters
+    ----------
+    image : nibabel.nifti1.Nifti1Image
+        NIFTI image to be closed loaded as a nibabel Nifti1Image object
+    iterations : int
+        number or closing iterations to perform on the input image
+
+    Returns
+    -------
+    nibabel.nifti1.Nifti1Image
+        NIFTI image with given iterations of closing on it
     """
-    Wrapper for scipy's binary_closing function to close
-    a NIFTI image with n iterations
-    """
+
+    # extract voxel intensity array
     image_data = image.get_fdata()
+    # extract the afffine matrix
     image_affine = image.affine
-    closing_test = ndimage.binary_closing(image_data, iterations=iterations).astype(int)
+    # scipy closing
+    closed_array = ndimage.binary_closing(image_data,
+                     iterations=iterations).astype(int)
 
-    closing_test = new_img_like(image, closing_test, image_affine)
+    closed_nifti_img = new_img_like(image, closed_array, image_affine)
 
-    return closing_test
+    return closed_nifti_img
 
 def to_T2space(t2_img, t1_img, output_dir):
+    """ Wrapper for pypreprocess's coregister function, used in T2 pipeline to
+    coregister T1 image to T2 image 
+
+    Parameters
+    ----------
+    t2_img : str
+        path for the t2 image
+    t1_img : str
+        path for the t1 image
+    output_dir : str
+        location to write the corregistered image
+
+    Returns
+    -------
+    dict
+        dictionary object consisting of all input and output info including the
```

Comment:
Then you probably want to state what are the keys of the dictionary.

Reply:
realised that this is just the `SubjectData()` object defined by pypreprocess, updating that